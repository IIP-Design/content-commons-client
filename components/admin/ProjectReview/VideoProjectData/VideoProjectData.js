import './VideoProjectData.scss';

import { Icon, Loader } from 'semantic-ui-react';

/**
 *
 * VideoProjectData
 *
 */
import React from 'react';
import gql from 'graphql-tag';
import { graphql } from 'react-apollo';
import { object } from 'prop-types';

const VideoProjectData = props => {
  const { error, loading, project } = props.data;

  if ( loading ) {
    return (
      <div
        style={ {
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'center',
          justifyContent: 'center',
          minHeight: '200px',
        } }
      >
        <Loader
          active
          inline="centered"
          style={ { marginBottom: '1em' } }
          content="Loading project data..."
        />
      </div>
    );
  }

  if ( error ) {
    return (
      <div className="video-project-data error">
        <p>
          <Icon color="red" name="exclamation triangle" />
          <span>Loading error...</span>
        </p>
      </div>
    );
  }

  if ( !project || !Object.keys( project ).length ) return null;

  const {
    projectTitle,
    author,
    team,
    visibility,
    categories,
    tags,
    descPublic,
    descInternal,
  } = project;

  return (
    <section className="section section--project_data">
      <h3 className="uppercase">Project Data</h3>
      <section className="project-data_meta section">
        <p>
          <b className="label">Video Title:</b>
          { ' ' }
          { projectTitle }
        </p>
        <p>
          <b className="label">Author:</b>
          { ' ' }
          { ( author && Object.keys( author ).length > 0 ) && `${author.firstName} ${author.lastName}` }
        </p>
        <p>
          <b className="label">Team:</b>
          { ' ' }
          { ( team && Object.keys( team ).length > 0 ) && team.name }
        </p>
        <p>
          <b className="label">Visibility Setting:</b>
          { ' ' }
          { visibility }
        </p>
      </section>

      <section className="project-data_taxonomy section">
        <p>
          <b className="label">Categories: </b>
          { ( categories && categories.length > 0 )
            ? categories
              .map( category => category.translations[0].name )
              .sort()
              .join( ', ' )
            : null }
        </p>
        <p>
          <b className="label">Tags: </b>
          { ( tags && tags.length > 0 )
            ? tags
              .map( tag => tag.translations[0].name )
              .sort()
              .join( ', ' )
            : null }
        </p>
      </section>

      <section className="project-data_description section">
        <p>
          <b className="label">Public Description:</b>
          { ' ' }
          { descPublic }
        </p>
        <p>
          <b className="label">Internal Description:</b>
          { ' ' }
          { descInternal }
        </p>
      </section>
    </section>
  );
};

VideoProjectData.propTypes = {
  data: object,
};

const VIDEO_PROJECT_REVIEW_DATA_QUERY = gql`
  query VideoProjectReviewData($id: ID!) {
    project: videoProject(id: $id) {
      id
      projectTitle
      author {
        id
        firstName
        lastName
      }
      team {
        id
        name
      }
      visibility
      descPublic
      descInternal
      categories {
        id
        translations {
          id
          name
        }
      }
      tags {
        id
        translations {
          id
          name
        }
      }
    }
  }
`;

export default graphql( VIDEO_PROJECT_REVIEW_DATA_QUERY, {
  options: props => ( {
    variables: {
      id: props.id,
    },
  } ),
} )( VideoProjectData );

export { VIDEO_PROJECT_REVIEW_DATA_QUERY };
