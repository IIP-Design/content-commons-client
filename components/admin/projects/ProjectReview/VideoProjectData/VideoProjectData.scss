.review-project .project-data {

  &_meta {

    p {
      line-height: 1.5285em;
    }
  }

  &_taxonomy {
    
    p {
      line-height: 1.5285em;
    }
  }

  &_description {
    
    p:first-child {
      margin-bottom: 1em;
    }
  } 
}