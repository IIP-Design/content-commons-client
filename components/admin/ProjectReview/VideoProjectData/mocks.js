import { VIDEO_PROJECT_REVIEW_DATA_QUERY } from './VideoProjectData';

export const props = { id: '123' };

export const mocks = [
  {
    request: {
      query: VIDEO_PROJECT_REVIEW_DATA_QUERY,
      variables: { id: props.id },
    },
    result: {
      data: {
        project: {
          id: '123',
          projectTitle: 'test title',
          author: {
            id: 'au438',
            firstName: 'Jane',
            lastName: 'Doe',
          },
          team: {
            id: 't11',
            name: 'IIP Video Production',
          },
          visibility: 'PUBLIC',
          descPublic: 'The public description',
          descInternal: 'The internal description',
          categories: [
            {
              id: 'c77',
              translations: [
                {
                  id: 'ct7',
                  name: 'about america',
                },
                {
                  id: 'ct8',
                  name: 'Amérique',
                },
              ],
            },
          ],
          tags: [
            {
              id: 't44',
              translations: [
                {
                  id: 'tt4',
                  name: 'american culture',
                },
                {
                  id: 'tt5',
                  name: 'Culture américaine',
                },
              ],
            },
          ],
        },
      },
    },
  },
];

export const errorMocks = [
  {
    ...mocks[0],
    result: {
      errors: [{ message: 'There was an error.' }],
    },
  },
];

export const nullMocks = [
  {
    ...mocks[0],
    result: {
      data: { project: null },
    },
  },
];

export const nullCatTagsMocks = [
  {
    ...mocks[0],
    result: {
      data: {
        project: {
          ...mocks[0].result.data.project,
          categories: null,
          tags: null,
        },
      },
    },
  },
];

export const emptyCatTagsMocks = [
  {
    ...mocks[0],
    result: {
      data: {
        project: {
          ...mocks[0].result.data.project,
          categories: [],
          tags: [],
        },
      },
    },
  },
];

export const nullAuthorTeamMocks = [
  {
    ...mocks[0],
    result: {
      data: {
        project: {
          ...mocks[0].result.data.project,
          author: null,
          team: null,
        },
      },
    },
  },
];

export const emptyAuthorTeamMocks = [
  {
    ...mocks[0],
    result: {
      data: {
        project: {
          ...mocks[0].result.data.project,
          author: {},
          team: {},
        },
      },
    },
  },
];
