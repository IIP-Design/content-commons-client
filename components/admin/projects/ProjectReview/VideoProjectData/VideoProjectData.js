/**
 *
 * VideoProjectData
 *
 */
import React from 'react';
import { object } from 'prop-types';
import gql from 'graphql-tag';
import { graphql } from 'react-apollo';

import './VideoProjectData.scss';

const VideoProjectData = props => {
  const { error, loading, project } = props.data;

  if ( loading ) return 'Loading the project...';
  if ( error ) return `Error! ${error.message}`;

  if ( !project || !Object.keys( project ).length ) return null;

  const {
    projectTitle,
    author,
    owner,
    visibility,
    categories,
    tags,
    descPublic,
    descInternal
  } = project;

  return (
    <section className="section section--project_data">
      <h3>PROJECT DATA</h3>
      <section className="project-data_meta section">
        <p><span className="label">Video Title:</span> { projectTitle }</p>
        <p><span className="label">Author:</span> { author }</p>
        <p><span className="label">Owner:</span> { owner }</p>
        <p><span className="label">Privacy Setting:</span> { visibility }</p>
      </section>

      { ( categories.length && tags.length )
        ? (
          <section className="project-data_taxonomy section">
            <p>
              <span className="label">Categories: </span>
              { categories
                .map( category => category.translations.name )
                .join( ', ' ) }
            </p>
            <p>
              <span className="label">Tags: </span>
              { tags
                .map( tag => tag.translations.name )
                .join( ', ' ) }
            </p>
          </section>
        ) : null }

      <section className="project-data_description section">
        <p><span className="label">Public Description:</span> { descPublic }</p>
        <p><span className="label">Internal Description:</span> { descInternal }</p>
      </section>
    </section>
  );
};

VideoProjectData.propTypes = {
  data: object
};

const VIDEO_PROJECT_REVIEW_DATA_QUERY = gql`
  query VideoProjectReviewData($id: ID!) {
    project: videoProject(id: $id) {
      id
      projectTitle
      author
      team {
        name
      }
      visibility
      descPublic
      descInternal
      categories {
        translations {
          name
        }
      }
      tags {
        translations {
          name
        }
      }
    }
  }
`;

export default graphql( VIDEO_PROJECT_REVIEW_DATA_QUERY, {
  options: props => ( {
    variables: {
      id: props.id
    },
  } )
} )( VideoProjectData );

export { VIDEO_PROJECT_REVIEW_DATA_QUERY };
