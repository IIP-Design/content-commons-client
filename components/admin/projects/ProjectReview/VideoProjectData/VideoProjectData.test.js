import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import wait from 'waait';
import { MockedProvider } from 'react-apollo/test-utils';
import { Icon, Loader } from 'semantic-ui-react';
import VideoProjectData, { VIDEO_PROJECT_REVIEW_DATA_QUERY } from './VideoProjectData';

const props = { id: '123' };

const mocks = [
  {
    request: {
      query: VIDEO_PROJECT_REVIEW_DATA_QUERY,
      variables: { id: props.id }
    },
    result: {
      data: {
        project: {
          id: '123',
          projectTitle: 'test title',
          author: {
            id: 'au438',
            firstName: 'Jane',
            lastName: 'Doe'
          },
          team: {
            name: 'IIP Video Production'
          },
          visibility: 'PUBLIC',
          descPublic: 'The public description',
          descInternal: 'The internal description',
          categories: [
            {
              translations: {
                name: 'about america'
              }
            },
            {
              translations: {
                name: 'Amérique'
              }
            }
          ],
          tags: [
            {
              translations: {
                name: 'american culture'
              }
            },
            {
              translations: {
                name: 'Culture américaine'
              }
            }
          ]
        }
      }
    }
  }
];

const Component = (
  <MockedProvider mocks={ mocks } addTypename={ false }>
    <VideoProjectData { ...props } />
  </MockedProvider>
);

describe( '<VideoProjectData />', () => {
  it( 'renders initial loading state without crashing', () => {
    const wrapper = mount( Component );
    const videoProjectData = wrapper.find( 'VideoProjectData' );
    const loader = (
      <Loader
        active
        inline="centered"
        style={ { marginBottom: '1em' } }
        content="Loading project data..."
      />
    );

    expect( videoProjectData.exists() ).toEqual( true );
    expect( videoProjectData.contains( loader ) ).toEqual( true );
    expect( toJSON( videoProjectData ) ).toMatchSnapshot();
  } );

  it( 'renders error message & icon if error is thrown', async () => {
    const errorMocks = [
      {
        request: {
          query: VIDEO_PROJECT_REVIEW_DATA_QUERY,
          variables: { id: props.id }
        },
        result: {
          errors: [{ message: 'There was an error.' }]
        }
      }
    ];

    const wrapper = mount(
      <MockedProvider mocks={ errorMocks }>
        <VideoProjectData { ...props } />
      </MockedProvider>
    );
    // wait for the data and !loading
    await wait( 0 );
    wrapper.update();

    const videoProjectData = wrapper.find( 'VideoProjectData' );
    const div = videoProjectData
      .find( 'div.video-project-data.error' );
    const icon = <Icon color="red" name="exclamation triangle" />;
    const span = <span>Loading error...</span>;

    expect( div.exists() ).toEqual( true );
    expect( videoProjectData.contains( icon ) )
      .toEqual( true );
    expect( videoProjectData.contains( span ) )
      .toEqual( true );
  } );

  it( 'renders `null` if project is `null`', async () => {
    const nullMocks = [
      {
        request: {
          query: VIDEO_PROJECT_REVIEW_DATA_QUERY,
          variables: { id: props.id }
        },
        result: {
          data: { project: null }
        }
      }
    ];

    const wrapper = mount(
      <MockedProvider mocks={ nullMocks }>
        <VideoProjectData { ...props } />
      </MockedProvider>
    );
    await wait( 0 );
    wrapper.update();

    const videoProjectData = wrapper.find( 'VideoProjectData' );

    expect( videoProjectData.html() ).toEqual( null );
  } );

  it( 'renders the final state', async () => {
    const wrapper = mount( Component );
    await wait( 0 );
    wrapper.update();

    const videoProjectData = wrapper.find( 'VideoProjectData' );

    expect( toJSON( videoProjectData ) ).toMatchSnapshot();
  } );
} );
