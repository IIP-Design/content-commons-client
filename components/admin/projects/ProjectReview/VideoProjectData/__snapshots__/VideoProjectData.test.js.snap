// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoProjectData /> renders initial loading state without crashing 1`] = `
<VideoProjectData
  data={
    Object {
      "error": undefined,
      "fetchMore": [Function],
      "loading": true,
      "networkStatus": 1,
      "refetch": [Function],
      "startPolling": [Function],
      "stopPolling": [Function],
      "subscribeToMore": [Function],
      "updateQuery": [Function],
      "variables": Object {
        "id": "123",
      },
    }
  }
  id="123"
>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "minHeight": "200px",
      }
    }
  >
    <Loader
      active={true}
      content="Loading project data..."
      inline="centered"
      style={
        Object {
          "marginBottom": "1em",
        }
      }
    >
      <div
        className="ui active text centered inline loader"
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      >
        Loading project data...
      </div>
    </Loader>
  </div>
</VideoProjectData>
`;

exports[`<VideoProjectData /> renders the final state 1`] = `
<VideoProjectData
  data={
    Object {
      "error": undefined,
      "fetchMore": [Function],
      "loading": false,
      "networkStatus": 7,
      "project": Object {
        "author": Object {
          "firstName": "Jane",
          "id": "au438",
          "lastName": "Doe",
        },
        "categories": Array [
          Object {
            "translations": Object {
              "name": "about america",
            },
          },
          Object {
            "translations": Object {
              "name": "Amérique",
            },
          },
        ],
        "descInternal": "The internal description",
        "descPublic": "The public description",
        "id": "123",
        "projectTitle": "test title",
        "tags": Array [
          Object {
            "translations": Object {
              "name": "american culture",
            },
          },
          Object {
            "translations": Object {
              "name": "Culture américaine",
            },
          },
        ],
        "team": Object {
          "name": "IIP Video Production",
        },
        "visibility": "PUBLIC",
      },
      "refetch": [Function],
      "startPolling": [Function],
      "stopPolling": [Function],
      "subscribeToMore": [Function],
      "updateQuery": [Function],
      "variables": Object {
        "id": "123",
      },
    }
  }
  id="123"
>
  <section
    className="section section--project_data"
  >
    <h3
      className="uppercase"
    >
      Project Data
    </h3>
    <section
      className="project-data_meta section"
    >
      <p>
        <b
          className="label"
        >
          Video Title:
        </b>
         
        test title
      </p>
      <p>
        <b
          className="label"
        >
          Author:
        </b>
         
        Jane Doe
      </p>
      <p>
        <b
          className="label"
        >
          Team:
        </b>
         
        IIP Video Production
      </p>
      <p>
        <b
          className="label"
        >
          Privacy Setting:
        </b>
         
        PUBLIC
      </p>
    </section>
    <section
      className="project-data_taxonomy section"
    >
      <p>
        <b
          className="label"
        >
          Categories: 
        </b>
        about america, Amérique
      </p>
      <p>
        <b
          className="label"
        >
          Tags: 
        </b>
        american culture, Culture américaine
      </p>
    </section>
    <section
      className="project-data_description section"
    >
      <p>
        <b
          className="label"
        >
          Public Description:
        </b>
         
        The public description
      </p>
      <p>
        <b
          className="label"
        >
          Internal Description:
        </b>
         
        The internal description
      </p>
    </section>
  </section>
</VideoProjectData>
`;
